pkg/controller/replication/replication_controller.go

```diff
@@ -242,6 +220,7 @@ func (rm *ReplicationManager) Run(workers int, stopCh <-chan struct{}) {
 	go rm.podController.Run(stopCh)
 	for i := 0; i < workers; i++ {
 		go wait.Until(rm.worker, time.Second, stopCh)
+		go wait.Until(rm.abandoner, 0, stopCh)
```

Comment:
Yeah, I found the race to be difficult to handle. However, even if abandoner is part of the worker, the order of work is still non-deterministic, because *i)* we have multiple workers, and *ii)* we need to enqueue an abandoning work again if the patch fails.